### PR TITLE
Implement std::iter::Extend for glib::List

### LINKED ISSUE #1468: [FEATURE REQUEST] Implement `std::iter::Extend` for `glib::List`
A standard library has some useful functions which require `Extend`. E.g. `std::iter::unzip`, `std::iter::partition` and some other like `std::iter::collect_into` in nightly.

### PR DESCRIPTION
**Pull request by andy128k**
Closes #1468

### CODE REVIEW

File: glib/src/collections/list.rs [hunk 1]
```diff
@@ -486,6 +486,25 @@ impl<T: TransparentPtrType> std::iter::IntoIterator for List<T> {
     }
 }
 
+impl<T: TransparentPtrType> std::iter::Extend<T> for List<T> {
```
Comment (sdroege) [inline glib/src/collections/list.rs:489]: Could you add the same for `SList`? Basically same implementation

File: glib/src/collections/list.rs [hunk 2]
```diff
@@ -486,6 +486,25 @@ impl<T: TransparentPtrType> std::iter::IntoIterator for List<T> {
...
+    #[inline]
+    fn extend<I: IntoIterator<Item = T>>(&mut self, iter: I) {
+        let list: Self = iter.into_iter().collect();
```
Comment (sdroege) [inline glib/src/collections/list.rs:492]:
```suggestion
        let list = iter.into_iter().collect::<Self>();
```

File: glib/src/collections/slist.rs [hunk 3]
```diff
@@ -480,6 +480,25 @@ impl<T: TransparentPtrType> std::iter::IntoIterator for SList<T> {
     }
 }
 
+impl<T: TransparentPtrType> std::iter::Extend<T> for SList<T> {
+    #[inline]
+    fn extend<I: IntoIterator<Item = T>>(&mut self, iter: I) {
+        let list: Self = iter.into_iter().collect();
```
Comment (sdroege) [inline glib/src/collections/slist.rs:486]:
```suggestion
        let list = iter.into_iter().collect::<Self>();
```

File: glib/src/collections/list.rs (hunk 1 above)
Comment (andy128k) [reply glib/src/collections/list.rs:489]: Done

File: glib/src/collections/slist.rs (hunk 3 above)
Comment (andy128k) [reply glib/src/collections/slist.rs]: Done

**Review by sdroege — APPROVED** [review]

**Pull request merged**